File: Day28/SubarrayLCM.py
class Solution(object):
    def subarrayLCM(self, nums, k):
        count = 0
        n = len(nums)
        for i in range(n):
            curr = nums[i]
            if curr > k or k % curr != 0:
                continue
            for j in range(i, n):
                if i != j:
                    a, b = curr, nums[j]
                    while b != 0:
                        a, b = b, a % b
                    curr = (curr * nums[j]) // a
                if curr == k:
                    count += 1
                elif curr > k:
                    break
        return count

if __name__ == "__main__":
    nums = list(map(int, input("Enter array elements separated by space:\n").split()))
    k = int(input("Enter k:\n"))
    sol = Solution()
    print("Number of subarrays with LCM equal to k:", sol.subarrayLCM(nums, k))

File: Day28/SubarrayLCM.java
import java.util.*;

class SubarrayLCM {
    public int subarrayLCM(int[] nums, int k) {
        int count = 0, n = nums.length;
        for (int i = 0; i < n; i++) {
            int curr = nums[i];
            if (curr > k || k % curr != 0) continue;
            for (int j = i; j < n; j++) {
                if (i != j) {
                    int a = curr, b = nums[j];
                    while (b != 0) {
                        int temp = b;
                        b = a % b;
                        a = temp;
                    }
                    curr = (curr * nums[j]) / a;
                }
                if (curr == k) count++;
                else if (curr > k) break;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter number of elements:");
        int n = sc.nextInt();
        int[] nums = new int[n];

        System.out.println("Enter array elements:");
        for (int i = 0; i < n; i++) nums[i] = sc.nextInt();

        System.out.println("Enter k:");
        int k = sc.nextInt();

        SubarrayLCM sol = new SubarrayLCM();
        System.out.println("Number of subarrays with LCM equal to k: " + sol.subarrayLCM(nums, k));
        sc.close();
    }
}
